ddtrace/span.py

```diff
@@ -35,6 +35,7 @@ class Span(object):
         '_context',
         '_finished',
         '_parent',
+        '__weakref__',
```

Comment:
It's required to store weak references for these instances. In a sense is a huge change in the API, but it's not changing anything about how the `Span` works. If we have other ideas we may change how to propagate the `Span`.